### DISCUSSION TITLE
filter using ctx query by modifying a controller need help with the syntax please

### DISCUSSION THREAD

**Discussion by mannp** [strapi/strapi#7313] [Legacy Topics]
Hi there

I am trying to follow the queries logic and am struggling;

Here > https://strapi.io/documentation/v3.x/concepts/queries.html#api-reference &
here > https://strapi.io/documentation/v3.x/guides/custom-data-response.html#example

I have changed the Articles controller to return only the 'published' articles based on the status attribute and it works great

I am now trying to use the same logic to change the Category controller to only return a Category if the associated post status is "published'.

I am not sure how to access this status attribute in the ctx.query below, as it is data from another controller....? 
How do I change that query to access the status field in the Articles controller please?

```
module.exports = {
  async find(ctx) {
    let entities;

    ctx.query = {
      ...ctx.query,
      status: 'published',
    };
    ....
}
```

My graphql responses as follows;
```
{
  "data": {
    "allStrapiCategory": {
      "edges": [
        {
          "node": {
            "id": "Category_1",
            "articles": [
              {
                "status": "published",
                "category": 2
              },
            ]
          }
        },
        {
          "node": {
            "id": "Category_2",
            "articles": [
              {
                "status": "published",
                "category": 2
              },
              {
                "status": "draft",
                "category": null
}
```

**Comment by lauriejim** [comment]
Hello! I'm not 100% sure to understand the query and result you attend.
And you please precise this sentence `I am now trying to use the same logic to change the Category controller to only return a Category if the associated post status is "published'.`

**Comment by mannp** [comment]
Hi @lauriejim thanks for taking the time to respond :)

I am trying to create a list of categories and that list should include only categories associated with published posts.

I have tried using elemMatch and I get a list of all categories regardless of whether they are associated with published articles.
```
  query CategoryTemplate($slug: String!) {
    strapiCategory(
      articles: { elemMatch: { status: { eq: "published" } } }
      slug: { eq: $slug }
    ) {
```
As the elemMatch did not work I thought I'd try this way, by changing the ctx query to only return categories associated with published articles,,,, but I do not know how to modify the category ctx query to do that....

```
    ctx.query = {
      ...ctx.query,
    articles:
      status: 'published',
    };
```

**Comment by mannp** [comment]
Resolved, used a different grapghql query.
